### ISSUE TITLE
Question about dirty checking

### ISSUE THREAD

**Issue by maxgrv** [aurelia/binding#125]
Inside a modal dialog we are binding to computed properties and aurelia falls back to dirty checking.

That would be fine as the dialog is usually open for short times, but we noticed that the dirty checking does not end when the dialog is closed. Is there a way to tell aurelia to stop checking properties?

**Comment by EisenbergEffect** [comment]
It all depends on how the dialog was implemented. It sounds like it's not properly tearing down its views. If that was being done, the binding would be deactivated and dirty checking would cease. Is this a custom dialog implementation you did, or are you using one of the plugins?

**Comment by maxgrv** [comment]
This is a custom implementation as plugins we tried did not fully support all the feature of the excellent durandal dialog plugin we are used to :)

We use a compose element (inside a repeat.for bound to a "stack" array to manage multiple levels of dialogs) which is bound to the desired dialog vm, then the dialog is closed and the compose element is destroyed (deleting the dialog object from the "stack"). Which is the correct way to dispose of VMs instantiated by a compose custom element?

Thanks

**Comment by EisenbergEffect** [comment]
It's possible this is a bug in compose then. Would it be possible for you to create a reproduction of this built on top of the navigation skeleton? I'd like to take a look at it to understand your usage better and see if there's an issue we need to address on our end.

**Comment by maxgrv** [comment]
OK, next week we will try to make a repro.

**Comment by maxgrv** [comment]
Trying to pull together a repro I discovered some funny things. The issue we have seems to have multiple causes, which should be filed in different issues I think but I will enumerate them here for completeness: 

1. Dirty checking is NOT turned off after deactivation if binding is using a value converter 

Step to repro: just use the plain vanilla navigation skeleton (14.0, I cannot get the 15.1 skeleton working on my machine), and see how many times fullname is checked after you navigate away to another viewmodel. Then take the "upper" off the binding and see the difference.

2. Validation plugin does not clear its observers after the VM is torn down (probably related to https://github.com/aurelia/validation/issues/11, pretty serious one IMHO)

3. If, inside the dialog view, we use a compose element like this:

          <compose compose.ref="composeVM" view-model.bind="moduleId" model.bind="vm"></compose>  

and before closing the dialog we set moduleId and vm to null the dirty checking persists. The dialog view is torn down as it is loaded by another compose element inside a repeat.for loop. I did not have time to pull together a repro for this but it is easily solved by not setting the properties to null before tearing down the outer compose.
